crates/router/src/connector/adyen/transformers.rs

```diff
@@ -289,139 +352,341 @@ impl TryFrom<&types::BrowserInformation> for AdyenBrowserInfo {
-                            .get_required_value("token")
-                            .change_context(errors::ConnectorError::RequestEncodingFailed)
-                            .attach_printable("No token passed")?,
-                    };
-                    Ok(AdyenPaymentMethod::Gpay(gpay_data))
-                }
-
-                api_enums::WalletIssuer::ApplePay => {
-                    let apple_pay_data = AdyenApplePay {
-                        payment_type,
-                        apple_pay_token: wallet_data
-                            .get_required_value("wallet_data")
-                            .change_context(errors::ConnectorError::RequestEncodingFailed)?
-                            .token
-                            .to_owned()
-                            .get_required_value("token")
-                            .change_context(errors::ConnectorError::RequestEncodingFailed)
-                            .attach_printable("No token passed")?,
-                    };
-                    Ok(AdyenPaymentMethod::ApplePay(apple_pay_data))
-                }
-                api_enums::WalletIssuer::Paypal => {
-                    let wallet = AdyenPaypal { payment_type };
-                    Ok(AdyenPaymentMethod::AdyenPaypal(wallet))
-                }
-            },
-            _ => Err(errors::ConnectorError::MissingRequiredField {
-                field_name: "payment_method",
-            }),
-        }?;
-
-        let browser_info = if matches!(item.auth_type, storage_enums::AuthenticationType::ThreeDs) {
-            item.request
-                .browser_info
-                .clone()
-                .map(|d| AdyenBrowserInfo::try_from(&d))
-                .transpose()?
-        } else {
-            None
-        };
+fn get_billing_address(item: &types::PaymentsAuthorizeRouterData) -> Option<Address> {
+    let address = item
+        .address
+        .billing
+        .as_ref()
+        .and_then(|billing| billing.address.as_ref());
+    let city = address.and_then(|address| address.city.clone());
+    let country = address.and_then(|address| address.country.clone());
+    let house_number_or_name = address.and_then(|address| address.line1.clone());
+    let postal_code = address.and_then(|address| address.zip.clone());
+    let state_or_province = address.and_then(|address| address.state.clone());
+    let street = address.and_then(|address| address.line2.clone());
+    Some(Address {
+        city,
+        country,
+        house_number_or_name,
+        postal_code,
+        state_or_province,
+        street,
+    })
+}
 
-        let additional_data = match item.request.capture_method {
-            Some(storage_models::enums::CaptureMethod::Manual) => Some(AdditionalData {
-                authorisation_type: AuthType::PreAuth,
-            }),
-            _ => None,
-        };
+fn get_delivery_address(item: &types::PaymentsAuthorizeRouterData) -> Option<Address> {
```

Comment:
Remove the Option Since this function will always return an address, instead of wrapping the address in some at the end